lib/rules/prefer-named-capture-group.js

```diff
@@ -17,6 +17,12 @@ const {
 } = require("@eslint-community/eslint-utils");
 const regexpp = require("@eslint-community/regexpp");
 
+//------------------------------------------------------------------------------
+// Typedefs
+//------------------------------------------------------------------------------
+
+/** @typedef {import("../types").Linter.LintSuggestion} SuggestionResult */
```

Comment:
I think `function suggestIfPossible()`, where this type is used, doesn't return an array of `Linter.LintSuggestion` (which are processed suggestions, where `fix` is `{ range, text }`) but an array of [`SuggestedEdit`](https://github.com/eslint/rewrite/blob/4ec089e5b60d64f695d09a973dbc4eb72026f112/packages/core/src/types.ts#L541C13-L541C26) (where `fix` is a function).

Reply:
Good catch, thanks! I replaced `LintSuggestion` with `SuggestedEdit` in e3757861b7fd5babfc82cad54ad1b56fd37dd3f2.